### DISCUSSION TITLE
I need having more control over `<EdgeAnchor />`s

### DISCUSSION THREAD

**Discussion by Alireza29675** [xyflow/xyflow#2439] [Ideas]
Based on what I see in the code, if the edge is updatable, [two `<EdgeAnchors />` will be added](https://github.com/wbkd/react-flow/blob/main/src/components/Edges/wrapEdge.tsx#L162-L179) at the same time in the `source` and `target` positions.

Currently, there is no control over these anchors; for example, if I want my edges to have just `target`-side I basically can't. Or, if I want to give some styles to these anchors, I can't properly select them using CSS without any dirty and unstable hacks.

I wanted to know about the community's ideas on this. What implementation do you suggest?

**Comment by moklick** [comment]
I never thought about this. We could give them class names then it would be easy to hide one or style them separately. What do you think about this?

**Reply by Alireza29675** [reply]
What about having separated class names for now? We can add `type` optional prop to `<EdgeAnchor />` and using it in an additional class name like `react-flow__edgeupdater-${type}` (As the same as what we have in `<Edge />`)

Then we will have these anchors with separated class names:

```ts
<EdgeAnchor type="source" position={sourcePosition} centerX={sourceX} centerY={sourceY} radius={edgeUpdaterRadius} />
// class names: react-flow__edgeupdater, react-flow__edgeupdater-source
// ...
<EdgeAnchor type="target" position={sourcePosition} centerX={sourceX} centerY={sourceY} radius={edgeUpdaterRadius} />
// class names: react-flow__edgeupdater, react-flow__edgeupdater-target
```

**Reply by moklick** [reply]
yep, that's what I thought. I'll add this!

**Reply by moklick** [reply]
I will add separate class names to the group element that wraps the `EdgeAnchor`.

**Reply by Alireza29675** [reply]
@moklick makes sense to me! Thanks 👍🏽

**Reply by moklick** [reply]
added in v11 - https://github.com/wbkd/react-flow/commit/809de82eb71694a3b121e0f74805e69673678ae5

**Comment by makhataibar** [comment]
Hi! What if I wanna handle events from EdgeAnchor?
